lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -295,7 +295,7 @@ static Value commonShflSync(Location loc, ConversionPatternRewriter &rewriter,
   auto *cOpr = builder.newConstantOperand(clamp);
   auto *maskOpr = builder.newConstantOperand("0xffffffff");
   shfl(dOpr, aOpr, bOpr, cOpr, maskOpr);
-  return builder.launch(rewriter, loc, val.getType(), false);
+  return builder.launch(rewriter, loc, val.getType());
```

Comment:
I think we should move to using the nvvm intrinsic instead, it would have the right semantic and doesn't require full side effects. 
I can clean it up later though if you don't feel like doing this in this PR.

Reply:
> I think we should move to using the nvvm intrinsic instead, it would have the right semantic and doesn't require full side effects.

I agree.

> I can clean it up later though if you don't feel like doing this in this PR.

Sure, we could have a separate PR that switch from ptx to NVVM for all shuffle instructions.
I may want to switch my workload to pipeline optimization asap. Please feel free to make a PR and I can help take a review, thanks!

Reply:
Oh, another benefits of doing it in a separate PR is to compare the generated code using nvvm with this PR to see if there's any perf regressions.

Reply:
Sounds good, I'll send a follow up PR once this one lands.

> I may want to switch my workload to pipeline optimization asap. Please feel free to make a PR and I can help take a review, thanks!

FYI @zahimoud is working on some heavy changes in the pipeliner.

Reply:
Yes, we had a discussion about it and will be collaborating on the work